frida_tools/tracer.py

```diff
@@ -918,6 +923,18 @@ def _sync_handlers(self, change_id):
                 self._notify_update(function, new_handler, handler_file)
 
 
+class OutputFile(object):
+    def __init__(self, filename):
+        self._fd = codecs.open(self.filename, 'wb', 'utf-8')
+
+    def append(self, message):
+        self._fd.write(message)
+        self._fd.flush()
+
+    def __del__(self):
```

Comment:
Should avoid relying on GC behavior and instead close explicitly:

```suggestion
    def close(self):
```

(Let's also move this method above `append()` as methods should be ordered: constructor, "destructor", <anything else>.)